agenda/presentation/src/main/java/com/pronoidsoftware/agenda/presentation/detail/components/event/photo/EventDetailPhotoHolder.kt

```diff
@@ -0,0 +1,136 @@
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import coil.compose.AsyncImage
+import com.pronoidsoftware.agenda.presentation.R
+import com.pronoidsoftware.core.presentation.designsystem.PlusIcon
+import com.pronoidsoftware.core.presentation.designsystem.TaskyLightBlue2
+import com.pronoidsoftware.core.presentation.designsystem.TaskyTheme
+
+@Composable
+private fun PhotoBorder(
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier,
+    color: Color = TaskyLightBlue2,
+    size: Dp = 60.dp,
+    strokeWidth: Dp = 2.dp,
+    cornerRadius: Dp = 5.dp,
+    content: @Composable () -> Unit = { },
+) {
+    val density = LocalDensity.current
+    val cornerRadiusPx = with(density) { cornerRadius.toPx() }
+    Box(
+        modifier = modifier
+            .size(size + strokeWidth)
+            .clip(RoundedCornerShape(cornerRadius))
+            .clickable {
+                onClick()
+            },
+        contentAlignment = Alignment.Center,
+    ) {
+        content()
+        Canvas(
+            modifier = Modifier.fillMaxSize(),
+        ) {
+            drawRoundRect(
+                color = color,
+                cornerRadius = CornerRadius(cornerRadiusPx, cornerRadiusPx),
+                style = Stroke(
+                    width = with(density) { strokeWidth.toPx() },
+                ),
+            )
+        }
+    }
+}
+
+@Composable
+fun AddPhotoButton(
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier,
+    addIcon: ImageVector = PlusIcon,
+    iconColor: Color = TaskyLightBlue2,
+) {
+    PhotoBorder(
+        onClick = onClick,
+        modifier = modifier,
+    ) {
+        Icon(
+            imageVector = addIcon,
+            contentDescription = stringResource(id = R.string.add_photos),
+            tint = iconColor,
+        )
+    }
+}
+
+@Composable
+fun PhotoThumbnail(photoId: PhotoId, onClick: () -> Unit, modifier: Modifier = Modifier) {
+    PhotoBorder(
+        modifier = modifier,
+        onClick = onClick,
+    ) {
+        when (photoId) {
+            is PhotoId.PhotoUri -> {
+                AsyncImage(
+                    model = photoId.uri,
+                    contentDescription = null,
+                    modifier = Modifier
+                        .fillMaxSize(),
+                    contentScale = ContentScale.Crop,
+                )
+            }
+
+            is PhotoId.PhotoResId -> {
+                Image(
+                    painter = painterResource(id = photoId.resId),
+                    contentDescription = null,
+                    modifier = Modifier
+                        .fillMaxSize(),
+                    contentScale = ContentScale.Crop,
+                )
+            }
```

Comment:
And in which scenario would you show a resource here? Aren't we always either showing a local content Uri or load it from a URL?

Reply:
In the prod app yes, but I was struggling to think of a good way to be able to display photos in previews to test that I'm creating these correctly. So I use resources for the previews